src/IECoreScene/MeshAlgoConnectedVertices.cpp

```diff
@@ -0,0 +1,87 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2018, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "IECoreScene/MeshAlgo.h"
+
+using namespace std;
+using namespace IECore;
+using namespace IECoreScene;
+
+pair<IntVectorDataPtr, IntVectorDataPtr> MeshAlgo::connectedVertices( const MeshPrimitive *mesh )
+{
+	size_t numVertices = mesh->variableData< V3fVectorData >( "P", PrimitiveVariable::Vertex )->readable().size();
+	const vector<int> &numVerticesPerFace = mesh->verticesPerFace()->readable();
+	const vector<int> &vertexIds = mesh->vertexIds()->readable();
+
+	vector< set<int> > neighbors( numVertices );
+
+	int currentVertOffset = 0;
+	for ( auto &&vertsPerFace : numVerticesPerFace )
```

Comment:
I'm also not a fan, but maybe that means I'm an old curmudgeon?

Reply:
I don't think this is the place for `&&`. It's a topic I still find confusing, but in this case I believe the `&&` is a [universal reference](https://stackoverflow.com/questions/13230480/what-does-auto-tell-us), rather than an rvalue reference, and is in effect just `int &` in this scenario. This is backed up by printing the type of `i` using the code [here](https://stackoverflow.com/questions/81870/is-it-possible-to-print-a-variables-type-in-standard-c). Assuming I'm correct, then using `&&` provides no benefits here, and certainly has the downside that it confuses the reader. We should keep it simple, and use `auto &`, or arguably just `auto`.

Reply:
Yep, to my understanding this should only be a universal reference and the `&&` does't give us any benefits over a normal lvalue reference here. I was just wondering whether that's the new standard way of writing range based loops. But yeah it makes sense to not confuscate the code if there's a simpler solution ;)